File: sample/dist2/java/src/main/java/org/yanex/hello/IJSII417Derived.java
package org.yanex.hello;

@javax.annotation.Generated(value = "jsii-pacmak/0.19.0 (build 5d7824d)", date = "2019-10-28T18:25:08.410Z")
public interface IJSII417Derived extends software.amazon.jsii.JsiiSerializable, org.yanex.hello.IJSII417PublicBaseOfBase {

    java.lang.String getProperty();

    void bar();

    void baz();

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    final static class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements org.yanex.hello.IJSII417Derived {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
            this.setObjRef(objRef);
        }

        @Override
        public java.lang.String getProperty() {
            return this.jsiiGet("property", java.lang.String.class);
        }

        @Override
        public java.lang.Boolean getHasRoot() {
            return this.jsiiGet("hasRoot", java.lang.Boolean.class);
        }

        @Override
        public void bar() {
            this.jsiiCall("bar", Void.class);
        }

        @Override
        public void baz() {
            this.jsiiCall("baz", Void.class);
        }

        @Override
        public void foo() {
            this.jsiiCall("foo", Void.class);
        }
    }
}

File: sample/dist2/kotlin/src/main/kotlin/hello/IJSII417Derived.kt
package org.yanex.hello

@javax.annotation.Generated(value = "jsii-pacmak/0.19.0 (build 5d7824d)", date = "2019-10-29T17:33:08.484Z")
interface IJSII417Derived: software.amazon.jsii.JsiiSerializable, org.yanex.hello.IJSII417PublicBaseOfBase {
    val propertyValue: kotlin.String

    fun bar() {
        this.jsiiCall("bar", kotlin.Any::class.java)
    }

    fun baz() {
        this.jsiiCall("baz", kotlin.Any::class.java)
    }

    internal class `Jsii$Proxy`: software.amazon.jsii.JsiiObject, org.yanex.hello.IJSII417Derived {
        override val hasRoot: kotlin.Boolean

        override val propertyValue: kotlin.String

        protected constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
            this.objRef = objRef
            this.hasRoot = this.jsiiCall("hasRoot", kotlin.Boolean::class.java)
            this.propertyValue = this.jsiiCall("property", kotlin.String::class.java)
        }

        fun foo() {
            this.jsiiCall("foo", kotlin.Any::class.java)
        }

        fun bar() {
            this.jsiiCall("bar", kotlin.Any::class.java)
        }

        fun baz() {
            this.jsiiCall("baz", kotlin.Any::class.java)
        }
    }
}
